client/lib/domains/dns/index.js

```diff
@@ -5,22 +5,23 @@ const includes = require( 'lodash/collection/includes' ),
 	mapValues = require( 'lodash/object/mapValues' ),
 	endsWith = require( 'lodash/string/endsWith' );
 
-function validateAllFields( fieldValues ) {
+function validateAllFields( fieldValues, domainName ) {
 	return mapValues( fieldValues, ( value, fieldName ) => {
 		const isValid = validateField( {
 			name: fieldName,
 			value: value,
 			type: fieldValues.type,
+			domainName: domainName
```

Comment:
I must admit I'm also not particularly fond of enforcing shorthands to be listed first.